lnwire/lnwire.go

```diff
@@ -292,17 +292,49 @@ func writeElement(w io.Writer, element interface{}) error {
 		}
 
 	case *net.TCPAddr:
-		var ip [16]byte
-		copy(ip[:], e.IP.To16())
-		if _, err := w.Write(ip[:]); err != nil {
-			return err
-		}
+		switch net := e.Network(); net {
```

Comment:
This won't work as expected for IPv6 addresses as they'll still always return "tcp" from the `Network()` method. 

I think instead, we might be able to use the [To4](https://golang.org/pkg/net/#IP.To4) method on the `IP` struct to discern if an address if IPv6 or not. The method will return `nil` if the address an IPv6 address.